source/Halibut.Tests/TraceLogFileLogger.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.IO;
+
+namespace Halibut.Tests
+{
+    public class TraceLogFileLogger : IDisposable
+    {
+        string tempFilePath = Path.GetTempFileName();
+        string testHash;
+
+        public void SetTestHash(string testHash)
+        {
+            this.testHash = testHash;
+        }
+        
+        public void WriteLine(string logMessage)
+        {
+            File.AppendAllLines(tempFilePath, new[] { logMessage });
+        }
+
+        public bool CopyLogFileToArtifacts()
+        {
+            var currentDirectory = Directory.GetCurrentDirectory();
+            var rootDirectory = new DirectoryInfo(currentDirectory).Parent.Parent.Parent.Parent.Parent;
```

Comment:
The `.Parent` train here isn't great. Is there not a more legit way of finding the artifact directory?

Reply:
Maybe the test context might have something for this?

Reply:
Yeah it's definitely yuck, I just don't know if there is a pre-defined field for "what is the 'project root directory' for this test".

I'll have another look and improve it if I can.

Reply:
Without looking hard, there is `TestContext.CurrentContext.WorkDirectory`. That might be helpful? (Or there might be something else in the context area?)

Reply:
`WorkDirectory` was the one I tried, it's basically the current dir and I still had to navigate 5 levels up via the Parent prop to get to the root.

Reply:
Yeah, [WorkDirectory defaults to the current dir if not explicitly configured](https://octopusdeploy.slack.com/archives/C047M75KF6C/p1697250813966669).

We could configure it via nuke or the TC build, but I feel like it's diminishing returns at this point.

Reply:
There's no neat way of getting this that I can see, but I'm going to add a comment explaining the expected directory structure, so if it ever changes in the future (unlikely) we can update the chain of Parents accordingly.